libsolidity/formal/CHC.cpp

```diff
@@ -99,6 +101,20 @@ void CHC::analyze(SourceUnit const& _source)
 		return;
 	}
 
+	if (
+		m_settings.solvers == SMTSolverChoice::ELD() &&
+		!ModelChecker::availableSolvers().eld
+	)
+	{
+		m_errorReporter.warning(
+			7267_error,
+			SourceLocation(),
+			"CHC analysis was not possible since Eldarica was the only"
+			" Horn solver enabled, but it was not found in the system."
+		);
+		return;
+	}
```

Comment:
Shouldn't we generally warn the user if a solver is selected but not available even when some other solver is? I'd understand not wanting to issue an error and fail but this is just a warning.

Reply:
I think this is how it's done already. It's done separately in each engine because they use different solvers.